### PR TITLE
deps: bump kubebuilder to 104672d

### PR DESCRIPTION
**Pull request by Adirio**
**Description of the change:**

deps: bump kubebuilder to kubernetes-sigs/kubebuilder@104672d, update where necessary
docs/: document service account featrue

**Motivation for the change:** recent changes in kubebuilder

/area dependency

**Checklist**

If the pull request includes user-facing changes, extra documentation is required:
- [ ] Add a new changelog fragment in `changelog/fragments` (see [`changelog/fragments/00-template.yaml`](https://github.com/operator-framework/operator-sdk/tree/master/changelog/fragments/00-template.yaml))
- [ ] Add or update relevant sections of the docs website in [`website/content/en/docs`](https://github.com/operator-framework/operator-sdk/tree/master/website/content/en/docs)

### CODE REVIEW

File: website/content/en/docs/cli/operator-sdk_alpha.md
Comment (camilamacedo86) [inline website/content/en/docs/cli/operator-sdk_alpha.md:33]: I think we might hide this one since it is in a very premature stage

Comment (camilamacedo86) [reply website/content/en/docs/cli/operator-sdk_alpha.md:33]: E.g done in the past : https://github.com/operator-framework/operator-sdk/blob/v0.18.x/cmd/operator-sdk/cli/legacy.go#L94-L100

Comment (Adirio) [reply website/content/en/docs/cli/operator-sdk_alpha.md]: Done

**Review by camilamacedo86 — APPROVED** [review]
Great 🥇

File: internal/cmd/operator-sdk/cli/cli.go
Comment (camilamacedo86) [inline internal/cmd/operator-sdk/cli/cli.go:111]: In KB we merged the config-gen an alpha, however,  It is in a very very very init stage, it is not compatible with the language plugins and it is not covered with tests/ci, has not all. features or docs. We just merge to make easier allow it to be improved. So, for SDK, IMO we must hide that.

**Review by camilamacedo86 — APPROVED** [review]
/lgtm

**Comment by jmrodri** [comment]
/hold
Just holding so I can take a look at it.

File: internal/cmd/operator-sdk/cli/cli.go
Comment (estroz) [reply internal/cmd/operator-sdk/cli/cli.go:111]: That's why it's in alpha. I'd like to have it added to `operator-sdk`.

File: internal/kubebuilder/cmdutil/cmdutil.go
Comment (estroz) [inline internal/kubebuilder/cmdutil/cmdutil.go:17]: Can this entire package not be removed?

Comment (Adirio) [reply internal/kubebuilder/cmdutil/cmdutil.go:17]: It will be removed with plugin phase 1.5 so it didnt make sense to expose it now and then remove it inmediately after

File: internal/cmd/operator-sdk/cli/cli.go
Comment (camilamacedo86) [reply internal/cmd/operator-sdk/cli/cli.go:111]: Yes. But we have alpha commands as well. I think we need to wait for a little before allowing it indeed for alpha in the SDK side. So, IMO we ough to move forward with it hidden and then, we have the options to choose when we will provide it. See that the alpha commands also are described in the docs CLI/ section as well. 

we cannot ensure anything about it so far. However, if you folks think that is fine provide that. Then, I am. OK with.

Comment (estroz) [reply internal/cmd/operator-sdk/cli/cli.go:111]: Hiding an alpha tool is counter-intuitive IMO. Don't you _want_ people to try out these alpha commands and give feedback? Anything that's marked 'alpha' is pretty clearly not stable.

Comment (Adirio) [reply internal/cmd/operator-sdk/cli/cli.go:111]: Want to mention that I hided it under request, I have no strong opinion here. Hiding means no docs and no exposure.

Comment (camilamacedo86) [reply internal/cmd/operator-sdk/cli/cli.go:111]: Hi @estroz,

>  Don't you want people to try out these alpha commands and give feedback?

Regards the config-gen I am not sure if it is in a stage that would be nice to offer to the SDK users and get feedback. It is exactly the point/motivation for I asked for @adirio to hide that. 

However, I am ok with the final decision too.

Comment (estroz) [reply internal/cmd/operator-sdk/cli/cli.go:111]: Will let someone else be the tiebreaker here. 

/ping @jmrodri @rashmigottipati

Comment (camilamacedo86) [reply internal/cmd/operator-sdk/cli/cli.go:111]: PS.: We can make it available any time that we wish, just by removing the //todo: code that hides it as well.

**Review by jmrodri — APPROVED** [review]
/lgtm

File: internal/cmd/operator-sdk/cli/cli.go
Comment (jmrodri) [reply internal/cmd/operator-sdk/cli/cli.go:111]: I'd say let's merge this PR as is, and we can decide why and how we want alpha config-gen later.

Comment (estroz) [reply internal/cmd/operator-sdk/cli/cli.go:111]: #4670

**Pull request merged**
